upup/pkg/fi/cloudup/azuretasks/vmscaleset.go

```diff
@@ -173,14 +177,15 @@ func (s *VMScaleSet) Find(c *fi.Context) (*VMScaleSet, error) {
 		StorageProfile: &VMScaleSetStorageProfile{
 			VirtualMachineScaleSetStorageProfile: profile.StorageProfile,
 		},
-		RequirePublicIP:    to.BoolPtr(ipConfig.PublicIPAddressConfiguration != nil),
-		SKUName:            found.Sku.Name,
-		Capacity:           found.Sku.Capacity,
-		ComputerNamePrefix: osProfile.ComputerNamePrefix,
-		AdminUser:          osProfile.AdminUsername,
-		SSHPublicKey:       sshKeys[0].KeyData,
-		Tags:               found.Tags,
-		PrincipalID:        found.Identity.PrincipalID,
+		RequirePublicIP:     to.BoolPtr(ipConfig.PublicIPAddressConfiguration != nil),
+		RequireLoadBalancer: to.BoolPtr(ipConfig.LoadBalancerBackendAddressPools != nil),
+		SKUName:             found.Sku.Name,
+		Capacity:            found.Sku.Capacity,
+		ComputerNamePrefix:  osProfile.ComputerNamePrefix,
+		AdminUser:           osProfile.AdminUsername,
+		SSHPublicKey:        sshKeys[0].KeyData,
+		Tags:                found.Tags,
+		PrincipalID:         found.Identity.PrincipalID,
```

Comment:
We will need to set the LoadBalancerName field in this Find function or else `kops update cluster` dry runs will always report trying to add a load balancer to the VMSS. Can we determine the load balancer name based on the `LoadBalancerBackendAddressPools` ?

Reply:
I don't believe azure supports multiple subnets yet so returning the first master subnet should work. 

I don't think we'll be able to get the name of the load balancer from the backend address pools. But since the name is just `api-<clustername>` i could write a function in azure_cloud.go file. Would you suggest writing a get tags function and then getting the cluster name that way or a cluster name function similar to the subscription id function?

Reply:
We need a way to determine the LoadBalancer's Name given the VMSS response object. I see below in `RenderAzure()` that the `ID` field in `LoadBalancerBackendAddressPools` resembles a URL path that includes the load balancer name. Does the VMSS object here contain that same path? If so we could parse the load balancer name out of the path.

Reply:
you were right in the first comment that we were able to get the load balancer name from the backend pools within the vmss profile. i setup something pretty much identical to the way kenji did the subnet id.